cli/azd/pkg/project/framework_service_npm.go

```diff
@@ -67,6 +65,35 @@ func (np *npmProject) Build(
 ) *async.TaskWithProgress[*ServiceBuildResult, ServiceProgress] {
 	return async.RunTaskWithProgress(
 		func(task *async.TaskContextWithProgress[*ServiceBuildResult, ServiceProgress]) {
+			// Exec custom `build` script if available
+			// If `build`` script is not defined in the package.json the NPM script will NOT fail
+			task.SetProgress(NewServiceProgress("Running NPM build script"))
+			if err := np.cli.RunScript(ctx, serviceConfig.Path(), "build", np.env.Environ()); err != nil {
+				task.SetError(err)
+				return
+			}
+
+			publishSource := serviceConfig.Path()
+
+			if serviceConfig.OutputPath != "" {
+				publishSource = filepath.Join(publishSource, serviceConfig.OutputPath)
+			}
+
+			task.SetResult(&ServiceBuildResult{
+				Restore:         restoreOutput,
+				BuildOutputPath: publishSource,
```

Comment:
This would also depend on the apps configuration.  If there is no output defined then is basically just a lint/compile in place and we give the user the service path back.

If they defined an output path we will append that to the service path.